### ISSUE TITLE
Unable to successfully zappa certify on windows

### ISSUE THREAD

**Issue by seanpaley** [Miserlou/Zappa#522]
Attempting to certify with either a local key file or one on s3.

First issue is an error like: No such file or directory: u'/tmp/account.key'
Created c:\tmp and got past that.

Now get
Calling certify for environment dev..
Certifying domain <domain> ..
OpenSSL Error: The filename, directory name, or volume label syntax is incorrect.

Failed to generate or install certificate! :(

**Comment by Miserlou** [comment]
Oh man, yeah. This is a pretty big one.

Windows is kind of a second class citizen until we can find somebody to help maintain support for that platform.

**Comment by seanpaley** [comment]
Well, for what it's worth I:
0. Installed GetGnuWin32 to get cat and openssl
1. created c:\tmp
2. made sure openssl was in my path
3. set OPENSSL_CONF to my openssl.cnf
4. edited all Popen calls in letsencrypt.py to pass command as string, not list.

That got me through. Waiting to see if it actually worked.

**Comment by Miserlou** [comment]
:-D!

**Comment by nueverest** [comment]
@seanpaley I got lost at step 3.

So I tried this:

- Download Git package for Windows here: https://git-scm.com/download/win
- Chose advanced options and installed Git Bash command line tools.
- Ran C:\Program Files\git-bash.exe
- Ran `openssl genrsa 2048 > account.key;`
- Activated Windows virtualenv where Zappa was installed. Use either `. Scripts/activate` or `source Scripts/activate` ([reference](http://stackoverflow.com/a/10451017/1783439))
- Ran `zappa certify`

I am still getting the original error: 

> OpenSSL Error: The filename, directory name, or volume label syntax is incorrect.


Any advice is welcome.

**Comment by seanpaley** [comment]
Hi @nueverest I actually switched back to using a mac, so I can't validate all the steps any more. Did you create a `c:\tmp` directory? The code expects that dir to exist and might be related to your issue.

**Comment by nueverest** [comment]
@seanpaley I just confirmed that `c:\tmp` exists.

**Comment by seanpaley** [comment]
Sorry. Going from memory.  #3 means to set an env variable OPENSSL_CONF to wherever your openssl.cnf is.  #4 is the painful one - I guess there is a difference with Popen on windows. The calls pass the command as a [], but I had to edit the file to pass as a string. Not sure if there was a better way but it worked.

**Comment by dan-nadler** [comment]
I was having the same issue. I followed all of @seanpaley suggestions, and that fixed it. I'm not sure which specifically fixed it, except I think that 4 was required. I did 1-3 with no luck. Then, I cloned the repo and changed the popen calls. That did it for me.

I am now getting "Domain challenge did not pass", but I'm assuming that it's user error on my end, or at least unrelated to this issue.
